examples/example_mnist_ae.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Google LLC
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""uses po2."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from collections import defaultdict
+
+import tensorflow.keras.backend as K
+from tensorflow.keras.datasets import mnist
+from tensorflow.keras.layers import Activation
+from tensorflow.keras.layers import Flatten
+from tensorflow.keras.layers import Input
+from tensorflow.keras.layers import *
+from tensorflow.keras.models import Model
+from tensorflow.keras.optimizers import Adam
+from tensorflow.keras.optimizers import SGD
+from tensorflow.keras.utils import to_categorical
+
+from qkeras import *
+from qkeras.utils import model_save_quantized_weights
+
+
+import numpy as np
+import tensorflow.compat.v1 as tf
+
+np.random.seed(42)
+
+NB_EPOCH = 100
+BATCH_SIZE = 64
+VERBOSE = 1
+NB_CLASSES = 10
+OPTIMIZER = Adam(lr=0.0001, decay=0.000025)
+VALIDATION_SPLIT = 0.1
+
+train = 1
+
+(x_train, y_train), (x_test, y_test) = mnist.load_data()
+
+RESHAPED = 784
+
+x_train = x_train.astype("float32")
+x_test = x_test.astype("float32")
+
+x_train = x_train[..., np.newaxis]
+x_test = x_test[..., np.newaxis]
+
+x_train /= 255.0
```

Comment:
shall we use 256 since it is powers-of-two number.

Reply:
This code is based on the other mnist examples and they all use 255. It should be the same, no?

Reply:
My thought: 256 is power of two and more hardware friendly for the implementation.